src/components/composer/composer/composer.ts

```diff
@@ -169,6 +167,17 @@ export class Composer extends Component<Props, SpreadsheetChildEnv> {
     return `width:${ASSISTANT_WIDTH}px;`;
   }
 
+  get nonHiddenFunctions(): { [key: string]: FunctionDescription } {
+    return Object.keys(functionRegistry.content)
+      .filter((key) => {
+        return !functionRegistry.content[key].hidden;
+      })
+      .reduce((prevContent, currentKey) => {
+        prevContent[currentKey] = functionRegistry.content[currentKey];
+        return prevContent;
+      }, {});
+  }
+
```

Comment:
Should I convert it into a state and use `useEffect` to update the state? Using `useEffect` seems to have less calculation.

Reply:
in practice I don't think it is a performance bottleneck to filter the functions every time.
There only a few hundred functions. Iterating that amount of items should be very fast